### PR TITLE
Flags disabling additional NodePorts and LoadBalancers added

### PR DESCRIPTION
**Pull request by Barteus**
Three flags were added to allow enabling the creation of the following:
- kubeflow load balancer
- kubeflow node port
- mlflow node port

For the purpose of backward compatibility, all flags are enabled by default.

Additionally, fixes applied to make CI work again:
- from PR: https://github.com/canonical/mlflow-operator/pull/72
- from PR: https://github.com/canonical/mlflow-operator/pull/73

### CODE REVIEW

**Comment by Barteus** [comment]
Closes: https://github.com/canonical/mlflow-operator/issues/11

File: charms/mlflow-server/tests/unit/test_operator.py [hunk 1]
```diff
@@ -373,3 +373,80 @@ def test_install_with_all_inputs(harness, mocker):
         mlflow_pod_defaults_minio_data["MLFLOW_TRACKING_URI"]
         == f"http://{harness.model.app.name}.{harness.model.name}.svc.cluster.local:{harness.charm.config['mlflow_port']}"  # noqa: E501
     )
+
+
+def test_install_without_nodeport_and_lb_services(harness, mocker):
+    harness.set_leader(True)
```
Comment (i-chvets) [inline charms/mlflow-server/tests/unit/test_operator.py:379]: Please add doc string indicating what is being tested, eg. `"""Test deployment without LB services."""`

File: charms/mlflow-server/src/charm.py [hunk 2]
```diff
@@ -162,63 +165,75 @@ def main(self, event):
...
-                        },
-                    ],
+                    "services": pod_spec_services,
                 },
             },
         )
         self.model.unit.status = ActiveStatus()
 
+    def _get_pod_spec_services(self, config):
+        pod_spec_services = []
```
Comment (i-chvets) [inline charms/mlflow-server/src/charm.py:175]: Please add doc sting describing what function does, eg. `"""Update pod spec services. Return updated pod spec."""`

**Review by i-chvets — APPROVED** [review]
Approved.

**Review by i-chvets — APPROVED** [review]
Approved.

**Pull request merged**